pydantic_settings/sources.py

```diff
@@ -374,7 +374,7 @@ class Settings(BaseSettings):
             args = get_args(annotation)
             if origin_is_union(get_origin(field.annotation)) and len(args) == 2 and type(None) in args:
                 for arg in args:
-                    if arg != type(None):
+                    if arg is not type(None):
```

Comment:
I think there's a `typing.NoteType` or something, might apply here. Fine as is though!

Reply:
we can simply use `if arq is not None`